document/js-api/index.bs

```diff
@@ -1095,28 +1097,23 @@ The algorithm <dfn>ToWebAssemblyValue</dfn>(|v|, |type|) coerces a JavaScript va
 1. If |type| is [=f64=],
     1. Let |f64| be ? [=ToNumber=](|v|).
     1. Return [=f64.const=] |f64|.
-1. Assert: |type| is a reference type.
```

Comment:
Keep this assertion?

Reply:
No particular objection, but could you clarify the point? There's no longer any shared prose that depends on it, and we need to handle all possible types explicitly anyway.

Reply:
If every case becomes guarded by an `if` then you may want to assert somehow that there are no other possibilities. But you're right, the assertion as is doesn't do that either. I don't have good off-hand suggestion how to fix that, though, except for keeping the original structure. Is there precedent for an "assert unreachable" in the end?

Reply:
I added an assertion. Maybe in the future we could move to a switch construct like in [DOM](https://dom.spec.whatwg.org/#dom-node-nodename); there's a few more places that could use it, though, so I'd be inclined to do it separately.

(I don't seem to have push access here, so please hit the "rebase" button for me.)